### PR TITLE
chore: fix master/agent Docker image vulnerabilities [DET-5926]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change installs `unattended-upgrades` and runs it once in the master and agent Docker images to ensure base OS packages are up to date.

## Test Plan
- [x] build images for master/agent
- [x] check no extra packages are pulled in (image size same as previous releases)
- [x] check affected packages are using fixed patch version listed in CVE.

## Commentary (optional)
Replaced upgrade of perl-base with this since this upgrades it.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by stoksc** [comment]
@ioga took another pass not using `unattended-upgrades`. It's 228MB (original) to 252MB (this) to 282MB (unattended), just on the size of `unattended` (confirmed this installs the same stuff, but it still installs packages we don't have). I'll take one last hack to make it better.

File: agent/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,14 @@
 FROM debian:10.3-slim
```
Comment (mackrorysd) [inline agent/Dockerfile:1]: Huh - I wonder why we're not using Ubuntu here? Nothing against Debian, but we're using Ubuntu pretty much everywhere else and it seems to be the best-supported by NVIDIA. I suspect they have more resources to be keeping Ubuntu base images, etc. up to date too. I wonder if we should switch.

Comment (stoksc) [reply agent/Dockerfile:1]: I was suggesting earlier we could at least use `debian:10-slim`, since it's up to date already.

Comment (stoksc) [reply agent/Dockerfile:1]: But @ioga mentioned some potential troubles with the same approach in envs.

Comment (ioga) [reply agent/Dockerfile:1]: we had the same issues with nvidia base images not having security upgrades couple months before. newer tags also didn't matter/help for nvidia. they just didn't update them on time.

Comment (stoksc) [reply agent/Dockerfile:1]: yep, confirmed latest release of `debian:10-slim` doesn't have the lz4 patch, so we'd need to do this anyway.

Comment (stoksc) [reply agent/Dockerfile:1]: ah but 10.10 does, I guess 10 != 10.10, which I assumed. oops.

File: master/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,13 @@ FROM debian:10.3-slim
 MAINTAINER Determined AI <hello@determined.ai>
 
 RUN apt-get update -yq \
-    && apt-get install -y ca-certificates perl-base \
+    && apt-get upgrade -s \
+        | grep "^Inst" \
+        | grep -e "Debian" -e "Debian-Security" \
+        | awk -F " " {'print $2'} \
+        | DEBIAN_FRONTEND=noninteractive xargs apt-get upgrade -yq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
```
Comment (stoksc) [inline master/Dockerfile:10]:
```suggestion
    && DEBIAN_FRONTEND=noninteractive apt-get install -yq \
```

**Review by ioga — APPROVED** [review]

**Comment by stoksc** [comment]
@mackrorysd i'll land this for now in the event we get busy, but i'll also look into just switching us to ubuntu in a bit.

**Pull request merged**